great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -50,9 +51,18 @@ def add(self, checkpoint: Checkpoint) -> Checkpoint:
 
         submit_event(
             event=CheckpointCreatedEvent(
-                checkpoint_id=persisted_checkpoint.id,
+                checkpoint_id=checkpoint.id,
                 validation_definition_ids=[
-                    vd.id for vd in persisted_checkpoint.validation_definitions
+                    validation_definition.id
+                    for validation_definition in checkpoint.validation_definitions
+                ],
+                actions=[
+                    ActionInfo(
+                        type=action.type,
+                        # notify_on is not a property of all Actions
+                        notify_on=getattr(action, "notify_on", None),
```

Comment:
I don't love this - thoughts?

Reply:
What would an alternative be?

Reply:
`isinstance` checks and some `mypy` ignores

Reply:
I think because this is analytics code and not part of the actual library, we're fine - `getattr` should be used sparingly

Reply:
I'm on team `isinstance` check.